packages/compass-components/src/components/radio-box-group.tsx

```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+import { RadioBoxGroup as LeafygreenRadioBoxGroup } from '@leafygreen-ui/radio-box-group';
+import { Label } from '@leafygreen-ui/typography';
+
+type BasicRadioGroupProps = React.ComponentProps<
+  typeof LeafygreenRadioBoxGroup
+>;
+
+interface RadioBoxGroupWithoutLabel extends BasicRadioGroupProps {
+  id?: string;
+  label: never;
+}
+
+interface RadioBoxGroupWithLabel extends BasicRadioGroupProps {
+  id: string; // We require an id for the label to reference.
+  label: string | JSX.Element;
+}
+
+type RadioBoxGroupProps = RadioBoxGroupWithoutLabel | RadioBoxGroupWithLabel;
+
+function RadioBoxGroup(props: RadioBoxGroupProps): JSX.Element {
```

Comment:
As you are handling the label in the component, you probably should extract it first from the rest of the props and be explicit in what cases you are passing it to the underlying component

```suggestion
function RadioBoxGroup({ label, id, ...props }: RadioBoxGroupProps): JSX.Element {
```

otherwise it's pretty easy to start passing some junk to the html elements:

<img width="360" alt="image" src="https://user-images.githubusercontent.com/5036933/154938493-561dcce9-729c-4b63-81e0-52767cc81915.png">

Reply:
Nice catch! Updated ty